src/main/kotlin/eu/long1/flutter/i18n/workers/I18nFileGenerator.kt

```diff
@@ -341,11 +337,23 @@ import 'package:flutter/material.dart';
 """
 
         private const val sClassHeader =
-            """class S implements WidgetsLocalizations {
+            """typedef void LocaleChangeCallback(Locale locale);
+
+class S implements WidgetsLocalizations {
   const S();
 
   static S current;
 
+  static Locale _locale;
+  static bool _shouldReload = false;
+
+  static set locale(Locale _newLocale) {
+    _shouldReload = true;
```

Comment:
This should be a clever setter: check that the new Locale is different the active one.

Reply:
OK with the clever refresh.
We should have something like that...
```
static set locale(Locale _newLocale) {
    if (getLang(S._locale) != getLang(_newLocale)) {
      _shouldReload = true;
      S._locale = _newLocale;
    }
  }
```

Reply:
That only checks the language, which is insufficient. The plugin supports also either country-based languages, or languages without a country. So we have to be backward-compatible. Check this method: `bool _isSupported(Locale, bool)`.

So maybe it's best to add an optional 2nd variable (bool withCountry)?

Reply:
The getLang method returns the correct result when locale is declared with country.
`getLang(new Locale("fr"))` returns "fr", and `getLang(new Locale("fr", "FR"))` returns "fr_FR", so I think that comparing `getLang(S._locale)` and `getLang(_newLocale)` is OK.

BTW, we could check if the new lang is supported, before deciding to change the locale.
```
  static set locale(Locale _newLocale) {
    if (S.delegate
            ._isSupported(_newLocale, _newLocale.countryCode?.isNotEmpty) &&
        getLang(S._locale) != getLang(_newLocale)) {
      _shouldReload = true;
      S._locale = _newLocale;
    }
  }
```

Reply:
Sounds like a good plan, could you commit the fixes and let's see where we stand?

Reply:
👍 Changes pushed.